tests/trestle/core/remote/cache_test.py

```diff
@@ -41,33 +42,42 @@ def as_file_uri(path: str) -> str:
     if platform.system() == const.WINDOWS_PLATFORM_STR:
         drive = pathlib.Path.cwd().resolve().drive
         return f'file:///{drive}{path}'
-    return f'file://{path}'
+    bare_path = path.lstrip('/')
+    return f'file:///{bare_path}'
 
 
-def test_fetcher_oscal(tmp_trestle_dir):
-    """Test whether fetcher can get an object from the cache as an oscal model."""
-    # Fetch from local content, expecting it to be cached and then fetched as an oscal model.
+def get_catalog_fetcher(tmp_trestle_dir: pathlib.Path,
+                        in_trestle: bool = False,
+                        relative: bool = False) -> Tuple[cache.FetcherFactory, Catalog, dict]:
+    """Instantiate a catalog and fetcher."""
     rand_str = ''.join(random.choice(string.ascii_letters) for x in range(16))
-    catalog_file = pathlib.Path(tmp_trestle_dir.parent / f'{rand_str}.json').__str__()
+    cat_name = f'{rand_str}.json'
+    dest_dir = tmp_trestle_dir / 'catalogs' if in_trestle else tmp_trestle_dir.parent
+    catalog_file = dest_dir / cat_name
     catalog_data = generators.generate_sample_model(Catalog)
-    catalog_data.oscal_write(pathlib.Path(catalog_file))
-    fetcher = cache.FetcherFactory.get_fetcher(pathlib.Path(tmp_trestle_dir), catalog_file)
-    # Create/update the cache copy
+    catalog_data.oscal_write(catalog_file)
+    saved_data = fs.load_file(pathlib.Path(catalog_file))
+    if relative:
+        catalog_str = f'./catalogs/{cat_name}' if in_trestle else f'../{cat_name}'
+    else:
+        catalog_str = str(catalog_file)
+    fetcher = cache.FetcherFactory.get_fetcher(tmp_trestle_dir, catalog_str)
+    return fetcher, catalog_data, saved_data
```

Comment:
I'm not sure why we are saving then loading data here. The extra read seems unnecessary (even if we need the file). Users can ask for the dict data by  OscalModel.dict(by_alias=True, exclude_unset=True)

Reply:
This was needed in order to test the raw mode of fetching - but I have removed the extra read and the corresponding test of raw read.